fix wordwrap

Bennylavaa opened the pull request:
Before:
![image](https://github.com/user-attachments/assets/b7da4f24-05f7-4299-bda1-95d73f9d3969)


After:
![image](https://github.com/user-attachments/assets/ff7e68a5-4e95-4c7f-be01-d02cc07fa5cb)

Bennylavaa committed:
fix wordwrap

Bennylavaa committed:
update version

Bennylavaa commented:
I also updated the version # ill leave that to you though seeing it looks like its never been updated.

Bennylavaa commented:
please wait on accepting this. I am looking into this commit https://github.com/jtuu/psochatlogaddon/commit/538f306642d275acbf93e76a6c2f1c6c411f00d5 

seems whatever he did broke highlighting because the name isnt matching as it expects. 

Say my name is Bennylava ingame its seeing it as EBennylav

Bennylavaa commented:
looks like it was fixed here https://github.com/jtuu/psochatlogaddon/pull/16

As for my word wrap it seems others are not a fan 

![image](https://github.com/user-attachments/assets/8f03302a-bb86-4c2a-91f0-53c69d1c4035)
is what they are wanting so I am not sure what I should do here seeing I think my colored names cause a conflict.

Bennylavaa committed:
fine tuning

- added a fixed for my wordwrap (will get another in the near future but with a setting for preference)
- fixed an issue for names larger than 7 characters (should now support all name lengths)
- changed the method for name coloring
- added highlight color setting.

Bennylavaa commented:
![image](https://github.com/user-attachments/assets/d8a8045d-ab3b-41b3-af7e-5634acb46129)

Ok this should be ready now. 

- added a fixed for my wordwrap (will get another in the near future but with a setting for preference)
- fixed an issue for names larger than 7 characters (should now support all name lengths)
- changed the method for name coloring
- added highlight color setting.

jtuu commented:
Sorry there's a new conflict.

Could you please explain what this does
https://github.com/jtuu/psochatlogaddon/blob/c5bd7589e05bb6c445a50052808de632fca8efad/Chatlog/init.lua#L461-L462

Bennylavaa commented:
Word wrap preferably done with spaces over mid word

jtuu commented:
I'm sorry but it seems like kind of a weird solution. Doesn't imgui already wrap text at word boundary? Are you sure this is the correct way to do it?

Bennylavaa committed:
Remove Conflict

Also removed the spacing formatting seeing imggui does it already I thought it did something differently.

Also fixed the highlight text not respecting any of the separator settings.

Bennylavaa committed:
Merge branch 'master' into master

Bennylavaa commented:
Fixed the conflict also removed the spacing thing, I didnt know is imgui did that correctly or not which after a few tests seems to work fine without it. 

I also corrected highlighted text not taking settings into account. 

This should really be my last PR for this I don't really have anything else to really add to this at this time. Maybe timestamp coloring but I think its fine with what is there atm.

jtuu commented:
Ok nice, thank you.
